### DISCUSSION TITLE
Tree shaking in NextJS 13 not working.

### DISCUSSION THREAD

**Discussion by woodjs** [chakra-ui/chakra-ui#7581] [General] [answered]
NextJS switched to SWC in version 13, but to be honest, it seems to me that tree-shaking did not work until version 13. I think it's a serious problem that deserves your attention.
What solutions are currently available?

nextJS has https://nextjs.org/docs/advanced-features/compiler#modularize-imports

**Comment by hassanzadeh-mj** [comment]
Considering the fact that the execution time of the components, especially in the app dir, is very high, I agree with this issue

**Comment by woodjs** [comment, marked as answer]
We discussed this on the ChakraUI Discord channel. As I understand it, the problem is in the "Framer Moution". I don’t know if the ChakraUI team is solving this problem, but it is known that the creator of this library is working on [ZagJS](https://zagjs.com/).

I can personally recommend: [RadixUI](https://www.radix-ui.com/), [MUI](https://mui.com/), [MantineUI](https://mantine.dev/), [TamagUI](https://tamagui.dev/)
